### PR TITLE
[SPARK-8373][PySpark]Add emptyRDD to pyspark and fix the issue when calling sum on an empty RDD

### PR DESCRIPTION
**Pull request by zsxwing**
This PR fixes the sum issue and also adds `emptyRDD` so that it's easy to create a test case.

### CODE REVIEW

**Comment by zsxwing** [comment]
retest this please

File: core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala [hunk 1]
```diff
@@ -425,6 +425,11 @@ private[spark] object PythonRDD extends Logging {
     iter.foreach(write)
   }
 
+  /** Create an RDD that has no partitions or elements. */
```
Comment (srowen) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: `emptyRDD[T]` should produce an RDD of `T`, not byte arrays right?

Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: T in the return type is not used. Actually, it can be an arbitrary type.

Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: I will update it to T

**Comment by davies** [comment]
LGTM

File: core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala [hunk 2]
```diff
@@ -425,6 +425,11 @@ private[spark] object PythonRDD extends Logging {
...
+  def emptyRDD[T](sc: JavaSparkContext): JavaRDD[T] = {
```
Comment (davies) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: Is this API used anywhere? If the answer is "no", we should remove it.

Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: I think this is just bringing it to parity with the scala SparkContext API, which we currently also only use in tests. This is probably fine.

Comment (davies) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: This class is `PythonRDD`, is not `JavaSparkContext`.

Comment (zsxwing) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: @davies right. I just sent #6867 to remove it.

Comment (andrewor14) [inline core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala]: Oh, I see

**Pull request closed without merging**